### PR TITLE
Property to control  inheritance of Layer Groups in virtual services

### PR DESCRIPTION
**Pull request by smithkm**
Also included a bean factory utility I had intended for a related fix that turned out to be impractical.  The factory class should be useful elsewhere though so I included it.

### CODE REVIEW

**Comment by aaime** [comment]
Just a quick note, when I read "inheritance" I was thinking of properties, or something like that, the idea that it meant making global groups visible in virtual services. Maybe it's just me and a better term is indeed not coming to mind, but wanted to report my "surprise", could have the same effect to others.

File: src/main/src/main/java/org/geoserver/ows/LocalWorkspaceCatalogFilter.java [hunk 1]
```diff
@@ -79,15 +80,35 @@ public boolean hideStyle(StyleInfo style) {
         return hideWorkspace(style.getWorkspace());
     }
 
+    /**
+     * Should local workspaces include layer groups from the global workspace
+     * @return
+     */
+    public static boolean workspaceLayerGroupInherit() {
+        String value = GeoServerExtensions.getProperty("GEOSERVER_GLOBAL_LAYER_GROUP_INHERIT");
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/ows/LocalWorkspaceCatalogFilter.java]: The property cannot really be changed at runtime can it? So why not put it in a static field and avoid the repeated lookups.

Comment (aaime) [inline src/main/src/main/java/org/geoserver/ows/LocalWorkspaceCatalogFilter.java]: Could have a package property setter for testing purposes

**Comment by aaime** [comment]
Would seem like a useful property to have in GeoServerInfo + UI, but I understand that takes more time, could be added later.

**Comment by aaime** [comment]
Hi Kevin, while not 100% sure I believe that while working on this pull request https://github.com/geoserver/geoserver/pull/1539 I actually made the global groups disappear from workspace specific services as a side effect.

**Comment by jodygarnett** [comment]
@smithkm any update here?

**Comment by bencaradocdavies** [comment]
Closing as inactive for eight months, now has merge conflicts with master, and even when it was submitted, failed to compile on Travis CI. Please revise and resubmit.

**Pull request closed without merging**